x-pack/plugins/transform/public/app/common/pivot_aggs.ts

```diff
@@ -238,6 +239,8 @@ export interface PivotAggsConfigWithExtra<T, ESConfig extends { [key: string]: a
   getAggName?: () => string | undefined;
   /** Helper text for the aggregation reflecting some configuration info */
   helperText?: () => string | undefined;
+  /** Provides error message type */
+  errorMessageType?: V | undefined;
```

Comment:
I reckon it's worth following the same pattern and adding the method `getErrorMessage` instead, which returns an array of errors, to make it future-proof. Then the side effect you have in `isValid` is not needed.

Reply:
Done in: [#a2f0df3](https://github.com/elastic/kibana/pull/197816/commits/a2f0df329ddc2cdb340e002a96ef1cf9573b4713)